consensus/tendermint/committee/committee_set_test.go

```diff
@@ -0,0 +1,374 @@
+		}
+	}
+
+	for _, size := range committeeSetSizes {
+		size := size
+		t.Run(fmt.Sprintf("committee set of %v member/s", size), func(t *testing.T) {
+			assertSet(t, size)
+		})
+	}
+
+	t.Run("cannot create empty set with members as nil", func(t *testing.T) {
+		_, err := NewSet(nil, common.Address{})
+		assertError(t, ErrEmptyCommitteeSet, err)
+	})
+
+	t.Run("cannot create empty set with members as types.Committee{}", func(t *testing.T) {
+		_, err := NewSet(types.Committee{}, common.Address{})
+		assertError(t, ErrEmptyCommitteeSet, err)
+	})
+}
+
+func TestSet_Size(t *testing.T) {
+	var committeeSetSizes = []int64{1, 2, 10, 100}
+	var assertSetSize = func(t *testing.T, n int64) {
+		committeeMembers := createTestCommitteeMembers(t, n, genRandUint64(int(n), maxRange))
+		// only testing size so don't care about sorting
+		set, err := NewSet(committeeMembers, committeeMembers[0].Address)
+		assertNilError(t, err)
+
+		setSize := set.Size()
+		if int64(setSize) != n {
+			t.Fatalf("expected committee set size: %v and got: %v", n, setSize)
+		}
+	}
+
+	for _, size := range committeeSetSizes {
+		size := size
+		t.Run(fmt.Sprintf("committee size of %v member/s", size), func(t *testing.T) {
+			assertSetSize(t, size)
+		})
+	}
+
+}
+
+func TestSet_Committee(t *testing.T) {
+	var committeeSetSizes = []int64{1, 2, 10, 100}
+	var assertSetCommittee = func(t *testing.T, n int64) {
+		committeeMembers := createTestCommitteeMembers(t, n, genRandUint64(int(n), maxRange))
+		set, err := NewSet(copyMembers(committeeMembers), committeeMembers[0].Address)
+		sort.Sort(committeeMembers)
+		assertNilError(t, err)
+
+		gotCommittee := set.Committee()
+
+		if !reflect.DeepEqual(committeeMembers, gotCommittee) {
+			t.Fatalf("expected committee: %v and got: %v", committeeMembers, gotCommittee)
+		}
+	}
+
+	for _, size := range committeeSetSizes {
+		size := size
+		t.Run(fmt.Sprintf("get committee of %v member/s", size), func(t *testing.T) {
+			assertSetCommittee(t, size)
+		})
+	}
+}
+
+func TestSet_GetByIndex(t *testing.T) {
+	committeeMembers := createTestCommitteeMembers(t, 4, genRandUint64(4, maxRange))
+	sort.Sort(committeeMembers)
+	set, err := NewSet(copyMembers(committeeMembers), committeeMembers[0].Address)
+	assertNilError(t, err)
+
+	t.Run("can get member by index", func(t *testing.T) {
+		expectedMember := committeeMembers[1]
+		gotMember, err := set.GetByIndex(1)
+		assertNilError(t, err)
+
+		if !reflect.DeepEqual(expectedMember, gotMember) {
+			t.Fatalf("expected member: %v and got %v", expectedMember, gotMember)
+		}
+	})
+
+	t.Run("error on accessing member index not in committee", func(t *testing.T) {
+		_, err := set.GetByIndex(6)
+		assertError(t, consensus.ErrCommitteeMemberNotFound, err)
+	})
+}
+
+func TestSet_GetByAddress(t *testing.T) {
+	committeeMembers := createTestCommitteeMembers(t, 4, genRandUint64(4, maxRange))
+	sort.Sort(committeeMembers)
```

Comment:
```suggestion
	sort.Sort(committeeMembers)
```

I don't think we need to sort here